src/averageAges.js

```diff
@@ -37,7 +44,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filterPeople = withChildren === undefined
+    ? people.filter(woman => woman.sex === 'f')
+    : people.filter(woman => {
+      return people.some(person => person.mother === woman.name);
+    });
+
+  const allWoman = filterPeople.map(woman => woman.died - woman.born);
+
+  const middleAge = allWoman.reduce((accumulator, curentValue) =>
```

Comment:
Use a more descriptive variable name instead of 'middleAge'. For example, 'averageAge'.